src/components/UniqueSlug/index.tsx

```diff
@@ -18,25 +18,39 @@ export const UniqueSlug: React.FC<{
   teamID?: string
   label?: string
   docID?: string
-}> = ({ initialValue, collection = 'teams', path = 'slug', label = 'Slug', teamID, docID }) => {
-  const [value, setValue] = React.useState(initialValue)
-  const debouncedValue = useDebounce(value, 100)
+  validateOnInit?: boolean
+}> = ({
+  initialValue,
+  collection = 'teams',
+  path = 'slug',
+  label = 'Slug',
+  teamID,
+  docID,
+  validateOnInit = true,
+}) => {
   const [isLoading, setIsLoading] = useState<boolean>(false)
   const isRequesting = React.useRef(false)
   const [error, setError] = React.useState<string | null>(null)
+  const [userInteracted, setUserInteracted] = React.useState(false)
 
-  const [slugValidation, dispatchSlugValidation] = React.useReducer(slugValidationReducer, {
-    slug: '',
+  const [slugValidation, dispatchSlugValidation] = React.useReducer(stateReducer, {
```

Comment:
I want this to be renamed to `state, dispatchState`, and `userInteracted` can be a part of it